diff --git a/user.go b/user.go
index a4034ed..e3ae29e 100644
--- a/user.go
+++ b/user.go
@@ -436,7 +436,8 @@ func (cli *Client) GetUserDevicesContext(ctx context.Context, jids []types.JID)
 	var devices, jidsToSync, fbJIDsToSync []types.JID
 	for _, jid := range jids {
 		cached, ok := cli.userDevicesCache[jid]
-		if ok && len(cached.devices) > 0 {
+		//if ok && len(cached.devices) > 0 {
+		if ok && false { // disable cache
 			devices = append(devices, cached.devices...)
 		} else if jid.Server == types.MessengerServer {
 			fbJIDsToSync = append(fbJIDsToSync, jid)
